dev/TEAL_opcodes.md

```diff
@@ -1128,3 +1229,74 @@ bitlen interprets arrays as big-endian integers, unlike setbit/getbit
 - Pushes: []byte
 - push a byte-array of length X, containing all zero bytes
 - LogicSigVersion >= 4
+
+## log
+
+- Opcode: 0xb0
+- Pops: *... stack*, []byte
+- Pushes: _None_
+- write bytes to log state of the current application
+- LogicSigVersion >= 5
+- Mode: Application
+
+`log` can be called up to MaxLogCalls times in a program, and log up to a total of 1k bytes.
+
+## tx_begin
+
+- Opcode: 0xb1
+- Pops: _None_
+- Pushes: _None_
+- Begin preparation of a new inner transaction
+- LogicSigVersion >= 5
+- Mode: Application
+
+## tx_field f
+
+- Opcode: 0xb2 {uint8 transaction field index}
+- Pops: *... stack*, any
+- Pushes: _None_
+- Set field F of the current inner transaction to X
+- LogicSigVersion >= 5
+- Mode: Application
+
+## tx_submit
+
+- Opcode: 0xb3
+- Pops: _None_
+- Pushes: _None_
+- Execute the current inner transaction. Panic on any failure.
```

Comment:
What happens if the same transaction is executed twice?
In Algorand, it would not be accepted because they would have the same transaction ID.
But what happens here?

Even more subtle: what happens if I execute twice the application in a group and each time the application issue the same payment transaction (same amount, same recipient).
Would this fail?
How would this affect composition? I think there are legit use cases where you want this to be authorized.

Reply:
It's fine.  No dupe detection.  Both are ok.

I feel that dupe detection is just for replay attacks, so only makes sense for signed transactions.  These aren't signed.  The reason they can't be "replayed" (in the bad sense) is because their outer transaction can't be replayed.  So no reason to refuse transactions that look the same.